File: Aula4-parte2/Em_python/Guerreiro.py
from Personagem import Personagem
class Guerreiro(Personagem):
    def __init__(self, nome: str, vida: int, nivel: int):
        super().__init__(nome, vida, nivel)
        self._defesa = 5
    @property
    def defesa(self):
        return self._defesa
    @defesa.setter
    def defesa(self, valor: int):
        if valor < 0:
            print("A defesa não pode ser negativa.")
        else:
            self._defesa = valor
    def ficha(self):
        print("Ficha do Guerreiro:")
        super().ficha()
        print(f"Defesa: {self._defesa}")

File: Aula4-parte2/Em_java/Guerreiro.java
public class Guerreiro extends Personagem{
    private int defesa;
    public Guerreiro(String nome, int vida, int nivel, int defesa){
        super(nome, vida, nivel);
        this.defesa = 5;
    }
    public int getDefesa() {
        return defesa;
    }
    public void setDefesa(int defesa){
        if(defesa >= 0){
            this.defesa = defesa;
        }else{
            System.out.println("Defesa não pode ser negativa.");
        }
    }
    @Override
    public void ficha(){
        System.out.println("Guerreiro");
        super.ficha();
        System.out.println("Defesa: " + this.defesa);
    }
}